eZ/Publish/Core/Repository/ContentService.php

```diff
@@ -588,22 +589,24 @@ public function createContent(APIContentCreateStruct $contentCreateStruct, array
                         $fieldDefinition,
                         $fieldValue
                     );
-                    if (!empty($fieldErrors)) {
-                        $allFieldErrors[$fieldDefinition->id][$languageCode] = $fieldErrors;
-                    }
-                }
-
-                if (!empty($allFieldErrors)) {
-                    continue;
                 }
 
-                $this->relationProcessor->appendFieldRelations(
+                $relationErrors = $this->relationProcessor->appendFieldRelations(
                     $inputRelations,
                     $locationIdToContentIdMapping,
                     $fieldType,
                     $fieldValue,
-                    $fieldDefinition->id
+                    $fieldDefinition
                 );
+
+                if (!empty($fieldErrors) || !empty($relationErrors)) {
+                    $allFieldErrors[$fieldDefinition->id][$languageCode] = array_merge($fieldErrors, $relationErrors);
+                }
+
+                if (!empty($allFieldErrors)) {
+                    continue;
```

Comment:
Not even a review, but my, when you're not familiar with the ContentService, it's hard to know what this actually continues from. That's a big foreach :-)